sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/xml/UDFXPathUtil.java

```diff
@@ -182,7 +182,7 @@ public long skip(long ns) throws IOException {
       // Bound skip by beginning and end of the source
       long n = Math.min(length - next, ns);
       n = Math.max(-next, n);
-      next += n;
+      next += (int)n;
```

Comment:
I'd put the cast on line 183 and make n an int. `length` and `next` are nonnegative ints.

Reply:
i see